### PR TITLE
docs: oauth2 provider migration

### PR DESCRIPTION
**Pull request by jannyHou**
After discussing with @raymondfeng and @deepakrkris we decided to update the documentations as this stage and postpone the migration guide. Thanks for the thoughtful suggestions!

Will close https://github.com/strongloop/loopback-next/issues/5184 after landing.

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [x] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [x] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: docs/site/migration/auth/oauth2.md [hunk 1]
```diff
@@ -6,12 +6,23 @@ sidebar: lb4_sidebar
...
+is used to setup a LoopBack application as an [OAuth 2.0](https://oauth.net/2/)
+server. It can be applied when people implement their LoopBack application as an
+OAuth 2.0 provider. For example, the API gateway application.
 
-{% include note.html content="
-This is a placeholder page, the task of adding content is tracked by the
-following GitHub issue:
-[loopback-next#3959](https://github.com/strongloop/loopback-next/issues/3959)
-" %}
+At this stage, authentication and authorization in LoopBack 4 focus on the
```
Comment (emonddr) [inline docs/site/migration/auth/oauth2.md:14]: > focus on the

focuses on the

File: docs/site/migration/auth/oauth2.md [hunk 2]
```diff
@@ -6,12 +6,23 @@ sidebar: lb4_sidebar
...
+communication to third-party servers like
+[google](https://developers.google.com/identity/protocols/oauth2) or
+[facebook](https://developers.facebook.com/docs/facebook-login/), instead of
+turning itself into an OAuth 2.0 provider. And given the complexity of
+[`loopback-component-oauth2`](https://github.com/strongloop/loopback-component-oauth2),
+we decide to postpone the migration guide to next stage.
+
+We have a simplified OAuth 2.0 server implementation in example
+[passport-login](https://github.com/strongloop/loopback-next/tree/master/examples/passport-login),
+which sets up an express application as an OAuth 2.0 provider. The provider is
```
Comment (emonddr) [inline docs/site/migration/auth/oauth2.md:24]: > The provider is created in file mock-oauth2-social-app.ts

Do you mean we `also` have another example of an oauth provider in mock-oauth2-social-app.ts ?

The way it is currently worded, it makes it sound as if mock-oauth2-social-app.ts is being used in examples/passport-login .

File: docs/site/migration/auth/oauth2.md [hunk 3]
```diff
@@ -6,12 +6,23 @@ sidebar: lb4_sidebar
...
+created in file
+[mock-oauth2-social-app.ts](https://github.com/strongloop/loopback-next/blob/master/extensions/authentication-passport/src/__tests__/acceptance/fixtures/mock-oauth2-social-app.ts).
+At this moment people who are interested to create such a provider can use it as
```
Comment (emonddr) [inline docs/site/migration/auth/oauth2.md:27]: > create such a provider can use it as

create such  providers can use them as


(this sentence only needs to change if we are talking about 2 distinct examples of an oauth provider)

File: docs/site/migration/auth/oauth2.md (hunk 1 above)
Comment (emonddr) [reply docs/site/migration/auth/oauth2.md:14]: @jannyHou , you are right `focuses` is fine.

File: docs/site/migration/auth/oauth2.md (hunk 2 above)
Comment (jannyHou) [reply docs/site/migration/auth/oauth2.md:24]: > mock-oauth2-social-app.ts is being used in examples/passport-login 

 @emonddr It is a fixture in the example's test folder. Hmm, is the meaning clear now?

File: docs/site/migration/auth/oauth2.md (hunk 3 above)
Comment (jannyHou) [reply docs/site/migration/auth/oauth2.md:27]: oh there is only one provider. Let me rephrase the sentence above.

File: docs/site/migration/auth/oauth2.md (hunk 1 above)
Comment (deepakrkris) [inline docs/site/migration/auth/oauth2.md:14]:
```suggestion
As of now, LoopBack 4 focuses on redirecting to third-party OAuth2 providers like google or facebook for authentication and authorization, instead of turning itself into an OAuth 2.0 provider.
```

File: docs/site/migration/auth/oauth2.md [hunk 4]
```diff
@@ -6,12 +6,23 @@ sidebar: lb4_sidebar
...
-{% include note.html content="
-This is a placeholder page, the task of adding content is tracked by the
-following GitHub issue:
-[loopback-next#3959](https://github.com/strongloop/loopback-next/issues/3959)
-" %}
+At this stage, authentication and authorization in LoopBack 4 focus on the
+communication to third-party servers like
+[google](https://developers.google.com/identity/protocols/oauth2) or
+[facebook](https://developers.facebook.com/docs/facebook-login/), instead of
+turning itself into an OAuth 2.0 provider. And given the complexity of
```
Comment (deepakrkris) [inline docs/site/migration/auth/oauth2.md:18]:
```suggestion
As of now, LoopBack 4 focuses on redirecting to third-party OAuth2 providers like google or facebook for authentication and authorization, instead of turning itself into an OAuth 2.0 provider.
```

File: docs/site/migration/auth/oauth2.md [hunk 5]
```diff
@@ -6,12 +6,23 @@ sidebar: lb4_sidebar
...
+[`loopback-component-oauth2`](https://github.com/strongloop/loopback-component-oauth2),
+we decide to postpone the migration guide to next stage.
```
Comment (deepakrkris) [inline docs/site/migration/auth/oauth2.md:20]:
```suggestion
we have decided to defer the migration guide for `loopback-component-oauth2`.
```

File: docs/site/migration/auth/oauth2.md [hunk 6]
```diff
@@ -6,12 +6,23 @@ sidebar: lb4_sidebar
...
+server. It can be applied when people implement their LoopBack application as an
+OAuth 2.0 provider. For example, the API gateway application.
 
-{% include note.html content="
-This is a placeholder page, the task of adding content is tracked by the
-following GitHub issue:
-[loopback-next#3959](https://github.com/strongloop/loopback-next/issues/3959)
-" %}
+As of now, LoopBack 4 focuses on redirecting to third-party OAuth2 providers
+like [google](https://developers.google.com/identity/protocols/oauth2) or
```
Comment (nabdelgadir) [inline docs/site/migration/auth/oauth2.md:15]: nitpick: `google` -> `Google`

File: docs/site/migration/auth/oauth2.md [hunk 7]
```diff
@@ -6,12 +6,23 @@ sidebar: lb4_sidebar
...
+[facebook](https://developers.facebook.com/docs/facebook-login/) for
```
Comment (nabdelgadir) [inline docs/site/migration/auth/oauth2.md:16]: nitpick: `facebook` -> `Facebook`

File: docs/site/migration/auth/oauth2.md [hunk 8]
```diff
@@ -6,12 +6,23 @@ sidebar: lb4_sidebar
...
+authentication and authorization, instead of turning itself into an OAuth 2.0
+provider. And given the complexity of
+[`loopback-component-oauth2`](https://github.com/strongloop/loopback-component-oauth2),
+we decide to defer the migration guide for it.
+
+We have a simplified OAuth 2.0 server implementation in extension
+[authentication-passport](https://github.com/strongloop/loopback-next/blob/master/extensions/authentication-passport)'s
+test fixtures, which sets up an express application as an OAuth 2.0 provider.
```
Comment (nabdelgadir) [inline docs/site/migration/auth/oauth2.md:24]: nitpick: `express` -> `Express`
